gef.py

```diff
@@ -9381,27 +9320,16 @@ class PatternSearchCommand(GenericCommand):
     _aliases_ = ["pattern offset",]
 
     @only_if_gdb_running
```

Comment:
I don't think we need an active debugging session to search, especially since now we have the default cycle length value. Not a big deal but it is more troublesome to have to start a debugging session just to search for a pattern (in particular when doing remote debugging).

Reply:
Agreed, but let's do that in another PR.

That way the generated change list shows it ;)

Reply:
We do need an active debugging session because we try do a smart eval of the pattern, so you can pass in registers, symbols, etc. 

But I agree with @Grazfather , that's out of scope of this PR.

Reply:
Can we maybe detect if it's running, and error out (with the same error as the decorator) if the target isn't running, but succeed if it's running _and_ there's nothing to evaluate?

Reply:
Done, it'll work regardless now as such

```python
        # 1. check if it's a symbol (like "$sp" or "0x1337")
        symbol = safe_parse_and_eval(pattern) if is_alive() else pattern
```